proto/bootz.proto

```diff
@@ -0,0 +1,167 @@
+  // BootstrappingData. The Bootz system will proceed to the next step if
+  // a SUCCESS is reported, otherwise it will retry or put the device in an
+  // out-of-service state. The device should validate the server's identity
+  // against the server_trust_cert it obtained in GetBootstrappingDataResponse.
+  rpc ReportStatus(ReportStatusRequest) returns (EmptyResponse) {}
+}
+
+// Initial request from device to the bootz-server. Contains a detailed
+// description of the chassis and its constituent control cards.
+message GetBootstrapDataRequest {
+  ChassisDescriptor chassis_descriptor = 1;
+  ControlCardState control_card_state = 2;
+  // A nonce that the bootstrap server should use when signing the response.
+  string nonce = 1001;
+}
+
+// Structured message containing the chassis details, and listing its component
+// control-card modules.
+message ChassisDescriptor {
+  string manufacturer = 1;
+  string part_number = 2;
+  // This field is populated for fixed form-factor chassis, and empty for
+  // modular chassis.
+  string serial_number = 3;
+  // For modular chassis this field will be used to list the control cards
+  // to be bootstrapped.
+  repeated ControlCard control_cards = 4;
+}
+
+// Details of the control card, including serial-number and the location
+// of the card on the chassis.
+message ControlCard {
+  string part_number = 1;
+  string serial_number = 2;
+  int32 slot = 3;
+}
+
+// The status of each control card (i.e. initialized or not). For first install
+// of the chassis, both control cards are in NOT_INSTALLED state. For 
+// install a hot-swapped standby, the status of the active card is INSTALLED, 
+// and that of the standby is NOT_INSTALLED. For FF chassis, the serial number
+// is that of the chassis, and status is NOT_INSTALLED.
+message ControlCardState {
+  string serial_number = 1;
+  ControlCardStatus status = 2;
+}
+
+enum ControlCardStatus {
+  CONTROL_CARD_STATUS_UNSPECIFIED = 0;
+  CONTROL_CARD_STATUS_NOT_INSTALLED = 1;
+  CONTROL_CARD_STATUS_INSTALLED = 2;
+}
+
+message GetBootstrapDataResponse {
+  message BootstrapDataResponse {
+    // The serial number of the control card to which this state should
+    // be applied.
+    string serial_num = 1;
+    // The device should download and install this image (or skip if the
+    // device is already running it).
+    SoftwareImage intended_image = 2;
+    // The device should set this hash as its Bootloader password.
+    string boot_password_hash = 3;
+    // This certificate should be used to validate the server when reporting
+    // progress.
+    string server_trust_cert = 4;
+    // Boot configuration is specified as structured data.
+    BootConfig boot_config = 5;
+    Credentials credentials = 6;
+    gnsi.pathz.v1.UploadRequest pathz = 7;
+    gnsi.authz.v1.UploadRequest authz = 8;
+    gnsi.certz.v1.UploadRequest certificates = 9;
+  } 
+  repeated BootstrapDataResponse responses = 1;
```

Comment:
What is the scenario where there is more than one response when GetBootstrapDataRequest only contains one serial number?  Is this to allow multiple server_trust_cert and/or multiple SoftwareImage (for multiple url's where it could be downloaded from)?

Reply:
the "active" sup registers both itself and any other sup's in the chassis